### DISCUSSION TITLE
Failed when connecting: Connection closed (code: 1006)

### DISCUSSION THREAD

**Discussion by draentropia** [roerohan/react-vnc#36] [Q&A] [answered]
Hi all,

I have a working VNC server (tested in a vncviewer) that I want to attach to my React application. However, after pasting the demo code and fixing the "ref" issue, I got the error `index.js:1 Failed when connecting: Connection closed (code: 1006)`

Does anybody know how can I fix this?

I'm using the version 0.5.1 of react-vnc and react-scripts 4.0.3 (I can't jump to 5 due to incompatibilities of another key component of the application).

Here is the code I'm using:

import React, { useRef } from 'react'
import { VncScreen } from 'react-vnc'

const Terminal = () => {
  const vncScreenRef = useRef(VncScreen)

  return (
    <VncScreen
      url='ws://192.168.60.111:5900'
      scaleViewport
      background="#000000"
      style={{
        width: '75vw',
        height: '75vh',
      }}
      ref={vncScreenRef}
    />
)}

Thanks!!
P.D: the code formatter is not working properly in my snipplet....

**Comment by draentropia** [comment, marked as answer]
I'd found a way to bypass the error. I had tightvnc and x11vnc in the VNC server (both works). Then I made a proxy between the pc with the react app and the vnc server using "noVNC". Finally, I pass to react-vnc the IP generated by the proxy.

**Reply by roerohan** [reply]
Yep, you'd need `noVNC` (or specifically `websockify`) to be running on the server, since that is the service which transforms the VNC stream to a `websocket` stream. The `react-vnc` library accepts a `ws://` endpoint, since browsers don't directly understand the `vnc` protocol.
